var/spack/repos/builtin/packages/msvc/package.py

```diff
@@ -57,6 +57,10 @@ def install(self, spec, prefix):
 
     @classmethod
     def determine_version(cls, exe):
+        if not exe.name.endswith(".exe") and not exe.name.endswith(".bat"):
```

Comment:
exe is not a pathlib object, just make this:
```suggestion
        if not exe.endswith(".exe") and not exe.endswith(".bat"):
```